packages/backend/server/src/core/config/resolver.ts

```diff
@@ -135,6 +159,52 @@ export class ServerConfigResolver {
   async initialized() {
     return this.server.initialized();
   }
+
+  @ResolveField(() => ReleaseVersionType, {
+    description: 'fetch latest available upgradable release of server',
+  })
+  async availableUpgrade(): Promise<ReleaseVersionType | null> {
+    const channel = RELEASE_CHANNEL_MAP.get(this.config.AFFINE_ENV) ?? 'stable';
+    const url = `https://affine.pro/api/worker/releases?channel=${channel}`;
+
+    try {
+      const response = await fetch(url, {
+        method: 'GET',
+        headers: {
+          Accept: 'application/json',
+          'Cache-Control': 'no-cache',
+        },
+      });
+
+      if (!response.ok) {
+        this.logger.error(
+          'failed to fetch affine releases',
+          await response.text()
+        );
+        return null;
+      }
+      const data = (await response.json()) as {
```

Comment:
The return is an array, should we take the first item?

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/hTwOityLamd4hitrae7M/037fcf9f-dbfa-40f2-b36e-e8c9a6f06a26.png)

Is the URL supposed to return the download address?

Reply:
whoops